In [1]:
%matplotlib inline
import pandas as pd

In [55]:
cast = pd.read_csv("C:/Users/rajhe/Downloads/cast.csv")
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4   NaN
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

### How many movies are listed in the titles dataframe?

In [115]:
len(cast)

3611212

Answer: 3611212 movies are listed in teh dataframe

### What are the earliest two films listed in the titles dataframe?

In [12]:
cast.sort_values(by='year').head(4)

title  year               name     type  \
2514470             Miss Jerry  1894    Blanche Bayliss  actress   
555929              Miss Jerry  1894     Chauncey Depew    actor   
464860              Miss Jerry  1894  William Courtenay    actor   
1718214  Soldiers of the Cross  1900          Reg Perry    actor   

                                                 character   n  
2514470               Miss Geraldine Holbrook (Miss Jerry) NaN  
555929   Himself - the Director of the New York Central... NaN  
464860                                        Mr. Hamilton NaN  
1718214                                               Lion NaN

Ans: Ealriest 2 films listed in dataframe are "Miss Jerry" and "Soldiers of teh Cross"
##### As title was repeating thrice had to take head(4)

### How many movies have the title "Hamlet"?

In [18]:
len(cast[cast.title=="Hamlet"])

313

### How many movies are titled "North by Northwest"?

In [20]:
len(cast[cast.title=="North by Northwest"])

112

### When was the first movie titled "Hamlet" made?

In [28]:
cast[cast.title=="Hamlet"].year.min()

1910

### List all of the "Treasure Island" movies from earliest to most recent.

In [33]:
(cast[cast.title=="Treasure Island"].sort_values(by='year'))

title  year                  name     type  \
830114   Treasure Island  1918        Charles Gorman    actor   
840745   Treasure Island  1918             Joe Grant    actor   
909157   Treasure Island  1918          Edwin Harley    actor   
3298306  Treasure Island  1918      Violet Radcliffe  actress   
1715014  Treasure Island  1918            Lloyd Perl    actor   
...                  ...   ...                   ...      ...   
2420089  Treasure Island  1999          Caveh Zahedi    actor   
1637661  Treasure Island  1999         Nick Offerman    actor   
1375015  Treasure Island  1999           J.P. Manoux    actor   
1797264  Treasure Island  1999  Victor Raider-Wexler    actor   
307749   Treasure Island  1999          Bob Byington    actor   

                       character     n  
830114           Prologue Player   NaN  
840745   Undetermined child role   NaN  
909157           Prologue Player   NaN  
3298306         Long John Silver   3.0  
1715014                Black Dog   4.0  
...                          ...   ...  
2420089                   Harold   9.0  
1637661                   Samuel   2.0  
1375015           Officer Hughes  18.0  
1797264                 Samowitz  10.0  
307749                    Thomas  12.0  

[190 rows x 6 columns]

### How many movies were made in the year 1950?

In [34]:
len(cast[cast.year==1950])

22006

### How many movies were made in the year 1960?

In [35]:
len(cast[cast.year==1960])

18456

### How many movies were made from 1950 through 1959?

In [40]:
len(cast[(cast.year>=1950) & (cast.year<=1959)])

210986

### In what years has a movie titled "Batman" been released?

In [42]:
cast[cast.title=="Batman"].year

95672      1943
96186      1943
114034     1989
119476     1989
149767     1989
           ... 
2907449    1989
3253550    1943
3343947    1989
3345872    1989
3356826    1989
Name: year, Length: 120, dtype: int64

### How many roles were there in the movie "Inception"?

In [52]:
len(cast[cast.title=="Inception"].character)

77

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [65]:
len(cast[(cast.title=="Inception") &(cast.n.isnull())])

26

### But how many roles in the movie "Inception" did receive an "n" value?

In [66]:
len(cast[(cast.title=="Inception")&(cast.n.notnull())])

51

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [76]:
cast[(cast.title=="North by Northwest")&(cast.n.notnull())].sort_values(by='n')

title  year                  name     type  \
840374   North by Northwest  1959            Cary Grant    actor   
3361396  North by Northwest  1959       Eva Marie Saint  actress   
1404704  North by Northwest  1959           James Mason    actor   
3024085  North by Northwest  1959   Jessie Royce Landis  actress   
343701   North by Northwest  1959        Leo G. Carroll    actor   
2923056  North by Northwest  1959  Josephine Hutchinson  actress   
1634552  North by Northwest  1959           Philip Ober    actor   
1229083  North by Northwest  1959         Martin Landau    actor   
2357939  North by Northwest  1959         Adam Williams    actor   
1746383  North by Northwest  1959          Edward Platt    actor   
642085   North by Northwest  1959     Robert Ellenstein    actor   
2212083  North by Northwest  1959          Les Tremayne    actor   
447893   North by Northwest  1959       Philip Coolidge    actor   
1454363  North by Northwest  1959         Patrick McVey    actor   
198496   North by Northwest  1959          Edward Binns    actor   
1334726  North by Northwest  1959             Ken Lynch    actor   

                           character     n  
840374            Roger O. Thornhill   1.0  
3361396                  Eve Kendall   2.0  
1404704              Phillip Vandamm   3.0  
3024085              Clara Thornhill   4.0  
343701                 The Professor   5.0  
2923056                Mrs. Townsend   6.0  
1634552              Lester Townsend   7.0  
1229083                      Leonard   8.0  
2357939                     Valerian   9.0  
1746383              Victor Larrabee  10.0  
642085                         Licht  11.0  
2212083                   Auctioneer  12.0  
447893                     Dr. Cross  13.0  
1454363               Sergeant Flamm  14.0  
198496                Captain Junket  15.0  
1334726  Charley - Chicago Policeman  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [78]:
cast[(cast.title=="Sleuth")&(cast.year==1972)].sort_values(by="n")

title  year                name     type  \
1644328  Sleuth  1972    Laurence Olivier    actor   
314691   Sleuth  1972       Michael Caine    actor   
360365   Sleuth  1972      Alec Cawthorne    actor   
1412647  Sleuth  1972  John (II) Matthews    actor   
2619278  Sleuth  1972  Eve (III) Channing  actress   
1396572  Sleuth  1972        Teddy Martin    actor   

                          character    n  
1644328                 Andrew Wyke  1.0  
314691                  Milo Tindle  2.0  
360365            Inspector Doppler  3.0  
1412647  Detective Sergeant Tarrant  4.0  
2619278             Marguerite Wyke  5.0  
1396572      Police Constable Higgs  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [82]:
cast[(cast.title=="Sleuth") &(cast.year==2007)&cast.n.notnull()].sort_values(by="n")

title  year           name   type    character    n
314692   Sleuth  2007  Michael Caine  actor       Andrew  1.0
1246855  Sleuth  2007       Jude Law  actor         Milo  2.0
1741124  Sleuth  2007  Harold Pinter  actor  Man on T.V.  3.0

### How many roles were credited in the silent 1921 version of Hamlet?

In [83]:
len(cast[(cast.title=="Hamlet")&(cast.year==1921)])

9

### How many roles were credited in Branagh’s 1996 Hamlet?

In [86]:
len(cast[(cast.title=="Hamlet")&(cast.name=="Kenneth Branagh")&(cast.year==1996)])

1

### How many "Hamlet" roles have been listed in all film credits through history?

In [88]:
len(cast[cast.character=="Hamlet"])

91

### How many people have played an "Ophelia"?

In [89]:
len(cast[cast.character=="Ophelia"])

111

### How many people have played a role called "The Dude"?

In [90]:
len(cast[cast.character=="The Dude"])

18

### How many people have played a role called "The Stranger"?

In [91]:
len(cast[cast.character=="The Stranger"])

208

### How many roles has Sidney Poitier played throughout his career?

In [92]:
len(cast[cast.name=="Sidney Poitier"])

43

### How many roles has Judi Dench played?

In [93]:
len(cast[cast.name=="Judi Dench"])

55

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [100]:
cast[(cast.name=="Cary Grant")&((cast.year>=1940)&(cast.year<=1949))&(cast.type=="Supporting actor")&(cast.n==2)].sort_values(by="year")

Empty DataFrame
Columns: [title, year, name, type, character, n]
Index: []

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [103]:
len(cast[((cast.year>=1940)&(cast.year<=1949))&(cast.name=="Cary Grant")])

23

### How many roles were available for actors in the 1950s?

In [105]:
len(cast[(cast.year//10==195)&(cast.type=="actor")])

154727

### How many roles were available for actresses in the 1950s?

In [107]:
len(cast[(cast.type=="actress")&(cast.year//10==195)])

56259

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [109]:
len(cast[(cast.n==1)&(cast.year>=1980)])

84264

### How many non-leading roles were available through from the beginning of film history through 1980?

In [113]:
len(cast[(cast.n.isnull())&(cast.year>=1980)])

945383

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [114]:
len(cast[(cast.year<=1980)&(cast.n.isnull())])

438086